default/python/AI/empire/colony_builders.py

```diff
@@ -2,8 +2,15 @@
 
 import AIDependencies
 from common.fo_typing import PlanetId, SpeciesName
+from common.listeners import register_post_handler, register_pre_handler
 from freeorion_tools import tech_is_complete
 from freeorion_tools.caching import cache_for_current_turn
+from freeorion_tools.lazy_initializer import InitializerLock
+
+_colony_builder_lock = InitializerLock("colony_builder")
+
+register_pre_handler("generateOrders", _colony_builder_lock.lock)
+register_post_handler("survey_universe", lambda *args: _colony_builder_lock.unlock())
```

Comment:
> Can't you change register_pre_handler / register_post_handler to accept a function instead of a string

I agree. In doubt, just internally convert to the function.__name__ if needed...

```
>>> def print_function_name(fun):
...     print(fun.__name__)
...
>>> print_function_name(print)
print
```

Reply:
> Phew, this looks almost like obfuscated python to me

Feel free to ask questions. 

> Can't you change register_pre_handler / register_post_handler to accept a function instead of a string? 

yes, it's possible.  Let's do it in a separate PR.